resources/assets/components/CampaignSingle/index.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import { flatMap, keyBy, map, sample, forEach, reject, filter } from 'lodash';
+
+import InboxItem from '../InboxItem';
+import ModalContainer from '../ModalContainer';
+import HistoryModal from '../HistoryModal';
+import PostFilter from '../PostFilter';
+import StatusCounter from '../StatusCounter';
+import { RestApiClient} from '@dosomething/gateway';
+
+class CampaignSingle extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const posts = keyBy(flatMap(props.signups, signup => {
+      return signup.posts.map(post => {
+        post.signup = signup;
+        return post;
+      });
+    }), 'id');
+
+    this.state = {
+      posts: posts,
+      filteredPosts: posts,
```

Comment:
If you keep track of `posts` and a string `filter` value, you can calculate `filteredPosts` in the render method rather than having to store two versions of the same thing in state. [Thinking In React](https://facebook.github.io/react/docs/thinking-in-react.html) is a good read on this.